node/src/components/consensus/era_supervisor.rs

```diff
@@ -420,9 +420,9 @@ where
         let validators: Validators<PublicKey> =
             validator_stakes.into_iter().map(scale_stake).collect();
 
-        let ftt = validators.total_weight()
-            * u64::from(self.highway_config().finality_threshold_percent)
-            / 100;
+        let total_weight = u128::from(validators.total_weight());
+        let ftt_percent = u128::from(self.highway_config().finality_threshold_percent);
+        let ftt = ((total_weight * ftt_percent / 100) as u64).into();
```

Comment:
If we did divide `finality_threshold_percent / 100` and then multiply by `validators.total_weight` we would get the same result, right?

Reply:
Almost: We'd lose some precision.
In the worst case, if all stakes are less than 100, we'd end up with a total weight of 0 because of that, and panic.

Reply:
I think the problem would have been that `finality_threshold_percent / 100` will always be 0.

Reply:
Ah, that way around! Yes, then everything would always be 0. :smile: